includes/amp-canonical-filters.php

```diff
@@ -0,0 +1,19 @@
+<?php
+
+// before Jetpack ever gets loaded, we need to remove a link rel prefetch for canonical AMP support
```

Comment:
Jetpack adds <a href="https://github.com/Automattic/jetpack/pull/1399/files#diff-9ba9fa5bb52bd6cf36c6c0bdd5ed4830R4939">a bunch of `dns-prefetch` tags</a> (not just for VideoPress) so a better approach might be to unhook it completely:
```
function amp_canonical_disable_jetpack_dns_fetch() {
    if ( class_exists( 'Jetpack' ) ) {
        remove_action( 'wp_head', array( 'Jetpack', 'dns_prefetch' ) ); 
    }
}
add_action( 'wp_head', 'amp_canonical_disable_jetpack_dns_fetch', 0 ); // Hook early so we can unhook Jetpack
```